### PR TITLE
LATIS-1061 munit migration

### PR DESCRIPTION
**Pull request by JakeMartin99**
In progress... upgrading tests to munit

### CODE REVIEW

**Comment by JakeMartin99** [comment]
Ok, this is rebased and good for review now

**Comment by JakeMartin99** [comment]
Ah, looks like I did miss a couple places where matchers were still being used. Am fixing now.

File: src/test/scala/latis/input/HapiJsonAdapterSuite.scala [hunk 1]
```diff
@@ -2,21 +2,17 @@ package latis.input
...
 import java.net.URI
 
-import org.scalatest.flatspec.AnyFlatSpec
-import org.scalatest.matchers.should.Matchers._
-import org.scalatest.Inside.inside
+import munit.CatsEffectSuite
 
-import latis.data._
-import latis.data.DomainData
+import latis.data.{DomainData, _}
```
Comment (lindholc) [inline src/test/scala/latis/input/HapiJsonAdapterSuite.scala:7]: Same as `import latis.data._`

File: src/test/scala/latis/input/HapiReaderSuite.scala [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+  test("read a dataset from a HAPI service") {
+    val uri = new URI("https://lasp.colorado.edu/lisird/hapi/info?id=nrl2_tsi_P1Y")
+
+    val ds = reader.read(uri)
+      .getOrElse(fail("Did not get dataset."))
+      .withOperation(Selection(id"time", Gt, "2010"))
+      .withOperation(Selection(id"time", Lt,  "2011"))
+
+    ds.id match {
+      case Some(id) => assertEquals (id, id"nrl2_tsi_P1Y")
```
Comment (lindholc) [inline src/test/scala/latis/input/HapiReaderSuite.scala:28]: Extra space here after `assertEquals`

File: build.sbt [hunk 3]
```diff
@@ -23,8 +23,9 @@ lazy val hapi = (project in file("."))
   
 lazy val commonSettings = Seq(
   libraryDependencies ++= Seq(
-    "com.typesafe"    % "config"      % "1.4.1",
-    "org.scalatest"  %% "scalatest"   % "3.2.9" % Test
+    "com.typesafe"                   % "config"               % "1.4.1",
+    "org.scalameta"                 %% "munit"                % "0.7.29" % Test,
+    "org.typelevel"                 %% "munit-cats-effect-3"  % "1.0.7" % Test
```
[the remark this answers is not in the document]
Comment (JakeMartin99) [reply build.sbt:28]: I was aligning it with the earlier lines (13-17), but I suppose that's not necessary

File: src/test/scala/latis/input/HapiJsonAdapterSuite.scala (hunk 1 above)
Comment (JakeMartin99) [reply src/test/scala/latis/input/HapiJsonAdapterSuite.scala:7]: Hmm, for some reason the import optimizer doesn't see it. Must be how it slipped by.

File: src/test/scala/latis/input/HapiReaderSuite.scala (hunk 2 above)
Comment (JakeMartin99) [reply src/test/scala/latis/input/HapiReaderSuite.scala:28]: Oops. My bad.

**Review by dlindhol — APPROVED** [review]

**Review by lindholc — APPROVED** [review]

**Pull request merged**